src/main/java/com/aws/greengrass/componentmanager/builtins/GreengrassRepositoryDownloader.java

```diff
@@ -206,8 +218,21 @@ private String getArtifactDownloadURL(ComponentIdentifier componentIdentifier, S
             }, "get-artifact-size", logger);
         } catch (InterruptedException e) {
             throw e;
+        } catch (GreengrassV2DataException e) {
+            if (e.statusCode() == HttpStatusCode.FORBIDDEN) {
+                throw new PackageDownloadException(getErrorString("Failed to get the pre-signed url of a public or a "
```

Comment:
do you need to specify Lambda here? Is there a way to know whether it is lambda or not before hand? I expect most customers wouldn't be using lambda so it seems odd to include in the error messages

This also seems very verbose. How about:
`Access denied when calling GetComponentVersionArtifact. Ensure certificate policy grants greengrass:GetComponentVersionArtifact`

Reply:
will update